toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -1332,11 +1412,46 @@ func (g *PkgGraph) fixIntraSpecCycle(trimmedCycle []*PkgNode) (err error) {
 	return
 }
 
-// fixPrebuiltSRPMsCycle attempts to fix a cycle if at least one node is a pre-built SRPM.
-// If a cycle can be fixed, edges representing the build dependencies of the pre-built SRPM will be removed.
-func (g *PkgGraph) fixPrebuiltSRPMsCycle(trimmedCycle []*PkgNode) (err error) {
+/*runNode is the currentNode which is being replaced in the cycle, and buildNode is its previous Node in the cycle*/
+func (g *PkgGraph) replaceCurrentRunNodeWithNewNode(runNode, newNode, buildNode *PkgNode) {
+	parentNodes := g.To(runNode.ID())
+	for parentNodes.Next() {
+		parentNode := parentNodes.Node().(*PkgNode)
+		if parentNode.Type == TypeLocalBuild && parentNode.SrpmPath == buildNode.SrpmPath {
+			g.RemoveEdge(parentNode.ID(), runNode.ID())
+
+			err := g.AddEdge(parentNode, newNode)
+			if err != nil {
+				logger.Log.Errorf("Adding edge failed for %v -> %v", parentNode, newNode)
+				return
+			}
+		}
+	}
+}
+
+func (g *PkgGraph) cloneAndReplaceRunToRemote(runNode *PkgNode, buildNodeToUpdate *PkgNode) {
+	//Mark node as unresolved remote to be fetched by pkgfetcher
+	upstreamAvailableNode := g.CloneNode(runNode)
+	upstreamAvailableNode.State = StateUnresolved
+	upstreamAvailableNode.Type = TypeRemoteRun
```

Comment:
We don't really want to clone the node in this case, because a remote unresolved run node does not have values for things like (S)RPM paths, a spec path, source path, etc. Adding this new node seems to be similar to how we create unresolved nodes in `grapher.go::addUnresolvedPackage()`:
```suggestion
	upstreamAvailableNode, err := g.AddPkgNode(runNode.VersionedPkg, StateUnresolved, TypeRemoteRun, "<NO_SRPM_PATH>", "<NO_RPM_PATH>", "<NO_SPEC_PATH>", "<NO_SOURCE_PATH>", "<NO_ARCHITECTURE>", "<NO_REPO>")
```

Currently, this will generate a duplicate error, but paired with changes suggested by @dmcilvaney the duplicates will be handled correctly.